wavefront/server/packages/flo_cloud/flo_cloud/aws/redshift.py

```
@@ -157,39 +158,135 @@ def execute_query(
+                projection,
+                where_clause,
+                limit,
+                offset,
+                order_by,
+                group_by=group_by,
+            )
+        else:
+            query = (
+                f'SELECT {projection} FROM {base_table} AS a '
+                f'WHERE {where_clause} {group_by_clause} {order_by_clause} '
+                f'LIMIT {limit} OFFSET {offset}'
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)